glew: Refactor as CMakePackage

ChristianTackeGSI opened the pull request:
In #18394 it was noted, that this package should be changed from a generic "Package" to a "CMakePackage". It makes a bunch of things easier.

And it uses all the common cmake code.

pinging @adamjstewart who asked about this conversion in https://github.com/spack/spack/pull/18394#issuecomment-683339463

adamjstewart commented on var/spack/repos/builtin/packages/glew/package.py:
```diff
@@ -15,22 +13,16 @@ class Glew(Package):
-                        (''), '../glew.pc')
+    def cmake(self, spec, prefix):
+        super(Glew, self).cmake(spec, prefix)
 
-            make()
-            make("install")
+        # https://github.com/Homebrew/legacy-homebrew/issues/22025
+        # Note: This file is generated only after cmake is run
+        filter_file(r'Requires: glu',
+                    (''), 'glew.pc')
```
I would do this like:
```python
@run_after('cmake')
def patch_glew_pc(self):
    filter_file('Requires: glu', '', 'glew.pc')
```

adamjstewart replied:
But keep the comment, that's useful

ChristianTackeGSI replied:
Will change.

BTW: What's the advantage of `run_after`? I personally consider the overriding and calling super even more readable, because it's more explicit.

adamjstewart replied:
`super` is certainly more Pythonic, `run_after` is more Spack-like. I'm honestly not sure if there is an advantage, maybe @alalazo has some opinions on this?

alalazo replied:
> I personally consider the overriding and calling super even more readable, because it's more explicit.

Having a pipeline with decorators permits to keep different operations well separated in different methods and doesn't impose any knowledge of MRO or calling convention of `super` (which should not be necessary if writing a package recipe).

ChristianTackeGSI replied:
Thanks for the explanation!

ChristianTackeGSI committed:
glew: Refactor as CMakePackage

In #18394 it was noted, that this package should be changed
from a generic "Package" to a "CMakePackage".
It makes a bunch of things easier.

And it uses all the common cmake code.
